ui/src/app/devices/components/DeviceHeaderForms/AddDeviceForm/types.ts

```diff
@@ -0,0 +1,19 @@
+import type { Device, DeviceIpAssignment } from "app/store/device/types";
+import type { Domain } from "app/store/domain/types";
+import type { Zone } from "app/store/zone/types";
+
+export type AddDeviceInterface = {
```

Comment:
Do you think this could extend the interface type from `CreateParams` that you've updated in device/types/actions.ts?

Reply:
Done! Forgot that `NonNullable` was a thing otherwise I'd have used it before!